sympy/core/power.py

```diff
@@ -1757,7 +1757,7 @@ def mul(d1, d2):
             res += terms[e1]*inco*x**(ex)
 
         if not (e.is_integer and e.is_positive and (e*d - n).is_nonpositive and
-                res == _mexpand(self)):
+                res == mexpand(self)):
```

Comment:
What are the implications of this for performance?

Reply:
none -- there are two functions. `_mexpand` has been promoted to public as `mexpand` and a new function `mexpand_cse` is available but not imported by default.

Actually there will be some impact: expressions like those added will test very quickly because `check_sol` is using `mexpand_cse` now.